apisix/plugins/saml-auth.lua

```diff
@@ -0,0 +1,92 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core = require("apisix.core")
+local resty_saml = require("resty.saml")
+
+local is_resty_saml_init = false
+
+local lrucache = core.lrucache.new({
+    ttl = 300, count = 512
+})
+
+local schema = {
+    type = "object",
+    title = "work with route or service object",
+    properties = {
+        sp_issuer = { type = "string" },
+        idp_uri = { type = "string" },
```

Comment:
Should be added with strict verification, like`coree.schema_def.uri_def`

Reply:
@soulbird `uri_def` is not suitable in this case, because all uri related properties could be in either full qualified address with host or path only.

Reply:
It's just an example, I think it's a bit too rough to only limit the string.

Reply:
Can any value be suitable as the input? For example, is `""` suitable for sp_issuer?